src/Motions/Word.ts

```diff
@@ -8,45 +8,56 @@ export class MotionWord extends Motion {
     private wordDelta: MotionWordPosition;
     private useBoundaryIfChange = false;
 
-    static wordSeparators = './\\\\()"\'\\-:,.;<>~!@#$%^&*|+=\\[\\]{}`~?';
+    private static wordSeparators = './\\\\()"\'\\-:,.;<>~!@#$%^&*|+=\\[\\]{}`~?';
+    private static blankSeparators = '\\s';
+    
+    separators: string;
 
-    static nextStartOrBoundaryIfChange(): Motion {
-        const obj = MotionWord.nextStart();
+    constructor(options: {useBlankSeparatedStyle: boolean} = {useBlankSeparatedStyle: false}) {
+        super();
+
+        this.separators = options.useBlankSeparatedStyle
+            ? MotionWord.blankSeparators : MotionWord.wordSeparators;
+    }
+
+    static nextStartOrBoundaryIfChange(args: {useBlankSeparatedStyle: boolean} = {useBlankSeparatedStyle: false}): Motion {
```

Comment:
Oh, I didn't make it clear enough.
We're not using the function directly. We defined a general interface [Action](https://github.com/aioutecism/amVim-for-VSCode/blob/master/src/Actions/Action.ts) and use it like [this](https://github.com/aioutecism/amVim-for-VSCode/blob/master/src/Modes/Mode.ts#L120).
So, the validation of `args` get ignored and things may getting wired if we define the `args` [here](https://github.com/aioutecism/amVim-for-VSCode/blob/master/src/Mappers/SpecialKeys/Motion.ts#L26) like this:

```
{ keys: 'W', motionGenerators: [MotionWord.nextStartOrBoundaryIfChange], args: {someOtherParam: someValue} }
```

Besides the special situation here, nesting default value definition can also act unexpectedly when you write something like this:

```ts
function test(args: {isFoo: boolean, isBar?: boolean} = {isFoo: true, isBar: true}) {
    console.log(args.isFoo); // true
    console.log(args.isBar); // undefined
}
test({isFoo: false});
```